mcda/utils/utils_for_main.py

```diff
@@ -0,0 +1,1200 @@
+
+    mcda_with_uncert = MCDAWithRobustness(config, input_matrix, is_exact_pdf_mask, is_poisson_pdf_mask)
+    n_random_input_matrices = mcda_with_uncert.create_n_randomly_sampled_matrices()
+
+    if is_sensitivity == "yes":
+        n_normalized_input_matrices = parallelize_normalization(n_random_input_matrices, polar)
+    else:
+        n_normalized_input_matrices = parallelize_normalization(n_random_input_matrices, polar, f_norm)
+
+    args_for_parallel_agg = [(weights, normalized_indicators)
+                             for normalized_indicators in n_normalized_input_matrices]
+
+    if is_sensitivity == "yes":
+        all_indicators_scores = parallelize_aggregation(args_for_parallel_agg)
+    else:
+        all_indicators_scores = parallelize_aggregation(args_for_parallel_agg, f_agg)
+
+    for matrix in all_indicators_scores:
+        normalized_matrix = rescale_minmax(matrix)
+        all_indicators_scores_normalized.append(normalized_matrix)
+
+    all_indicators_scores_means, all_indicators_scores_stds = estimate_runs_mean_std(all_indicators_scores)
+    all_indicators_means_scores_normalized, all_indicators_scores_stds_normalized = \
+        estimate_runs_mean_std(all_indicators_scores_normalized)
+
+    ranks = all_indicators_scores_means.rank(pct=True)
+
+    _save_output_files(scores=None, normalized_scores=None,
+                       ranks=ranks,
+                       score_means=all_indicators_scores_means,
+                       score_stds=all_indicators_scores_stds,
+                       score_means_normalized=all_indicators_means_scores_normalized,
+                       iterative_random_w_score_means=None,
+                       iterative_random_w_score_means_normalized=None,
+                       iterative_random_w_score_stds=None,
+                       input_config=input_config,
+                       index_column_name=index_column_name, index_column_values=index_column_values)
+
+    _plot_and_save_charts(scores=None, normalized_scores=None,
+                          score_means=all_indicators_scores_means, score_stds=all_indicators_scores_stds,
+                          score_means_normalized=all_indicators_means_scores_normalized,
+                          iterative_random_w_score_means=None,
+                          iterative_random_w_score_stds=None,
+                          iterative_random_w_score_means_normalized=None,
+                          input_matrix=input_matrix, config=input_config,
+                          is_robustness_indicators=is_robustness_indicators)
+
+
+def _compute_scores_for_all_random_weights(indicators: dict, is_sensitivity: str,
+                                           weights: Union[List[str], List[pd.DataFrame], dict, None],
+                                           f_agg: str) -> tuple[Any, Any, Any, Any]:
+    """
+    Computes the normalized mean scores and std of the alternatives in the case of randomly sampled weights.
```

Comment:
Any comments for parameters and the return types?

Reply:
That is my thinking: every private function, e.g., `_compute_scores_for_all_random_weights` (every function whose name starts with "_") does not get a full docstring. The same applies for methods decorated with the `@static` decorator.